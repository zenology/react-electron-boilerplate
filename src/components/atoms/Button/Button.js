const Button = ({ className, disabled, onClick = () => {}, children }) => {
  return (
    <button
      className={className}
      css={{ color: 'red' }}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export default Button
