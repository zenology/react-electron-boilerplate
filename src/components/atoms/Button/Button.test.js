import { render } from '@testing-library/react'
import Button from './Button'

describe('Button Component', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>Test Text</Button>)

    expect(container).toMatchSnapshot()
  })

  it('should disable button correctly', () => {
    const { getByRole } = render(<Button disabled>Test Text</Button>)
    const button = getByRole('button')

    expect(button).toHaveAttribute('disabled')
  })

  it('can fired click event', () => {
    const mockFn = jest.fn()
    const { getByRole } = render(<Button onClick={mockFn}>Test Text</Button>)
    const button = getByRole('button')

    button.click()

    expect(mockFn).toBeCalled()
  })
})
