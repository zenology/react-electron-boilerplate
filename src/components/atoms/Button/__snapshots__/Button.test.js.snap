// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
.emotion-0 {
  color: red;
}

<div>
  <button
    class="emotion-0"
  >
    Test Text
  </button>
</div>
`;
